File: algorithm/src/main/java/baekjoon/p11109/Main.java
package baekjoon.p11109;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        while (t-- > 0) {
            System.out.println(solve(sc.nextInt(), sc.nextInt(), sc.nextInt(), sc.nextInt()));
        }
    }

    private static String solve(int d, int n, int s, int p) {
        int a = d + n * p;
        int b = n * s;

        if (a < b) {
            return "parallelize";
        } else if (a > b) {
            return "do not parallelize";
        } else {
            return "does not matter";
        }
    }

}

File: algorithm/src/main/java/baekjoon/p11109/Main.kt
package baekjoon.p11109

fun main(args: Array<String>) {
    repeat(readLine()!!.toInt()) {
        val inputs = readLine()!!.split(" ").map { it.toInt() }
        println(solve(inputs))
    }
}

private fun solve(inputs: List<Int>): String {
    val a = inputs[0] + inputs[1] * inputs[3]
    val b = inputs[1] * inputs[2]

    return when {
        a < b -> "parallelize"
        a > b -> "do not parallelize"
        else -> "does not matter"
    }
}